multi-educator/src/Dscp/MultiEducator/Web/Educator/Auth.hs

```diff
@@ -37,21 +41,35 @@ deriveJSON defaultOptions ''EducatorAuthData
 instance FromJWT EducatorAuthData
 instance ToJWT EducatorAuthData
 
-{-
-instance IsAuth EducatorAuth EducatorAuthResult where
-    type AuthArgs EducatorAuth = '[LookupEducator, JWTSettings]
+---------------------------------------------------------------------------
+-- Data types
+---------------------------------------------------------------------------
+
+-- | Custom authentication type for auth-servant
+data MultiEducatorAuth
+
+-- | Type that holds MultiEducator's public key
+newtype MultiEducatorPublicKey = MultiEducatorPublicKey PublicKey
+
+instance IsAuth MultiEducatorAuth EducatorAuthData where
+    type AuthArgs MultiEducatorAuth = '[MultiEducatorPublicKey]
     runAuth _ _ = multiEducatorAuthCheck
 
+instance IsClientAuth MultiEducatorAuth where
+    data ClientAuthData MultiEducatorAuth = MultiEducatorClientAuthData SecretKey
+    provideAuth req (MultiEducatorClientAuthData sk) = signRequestBasic sk req
+
 ---------------------------------------------------------------------------
 -- Helpers
 ---------------------------------------------------------------------------
 
--- | This function returns AuthCheck that checks the signature of the JWT.
-multiEducatorAuthCheck :: LookupEducator -> JWTSettings -> AuthCheck EducatorAuthResult
-multiEducatorAuthCheck (LookupEducator lookupEducator) jwtSettings = do
-    (JWTData login) <- jwtAuthCheck jwtSettings
-    liftIO (lookupEducator login) >>= \case
-        Just (key, db) -> return $ EducatorData key db
-        Nothing -> return $ EducatorNotFound
-
--}
+-- | Checks the signature of the JWT and returns an 'EducatorAuthData'
+multiEducatorAuthCheck :: MultiEducatorPublicKey -> AuthCheck EducatorAuthData
+multiEducatorAuthCheck (MultiEducatorPublicKey mpk) = do
+    (pk, payload) <- checkJWitness
```

Comment:
This only checks the JWS signature, but does not perform other checks typical for JWT, such as checking for expiry in `exp` claim. I really think that at least expiry check should be used. However, it's better to consult TMP team on which claims do they use for their own authorization processing, just to double-check.